lib/taiko.js

```diff
@@ -341,23 +355,52 @@ function getEventProxy(target) {
  */
 module.exports.client = () => clientProxy;
 
+/**
+ * This will return a new RegExp Object
+ *
+ * @returns {Object}
+ */
+module.exports.regex = async regexString => {
+  var isValid = true;
+  var myRegexObject = null;
+  try {
+    myRegexObject = new RegExp(regexString);
+  } catch (e) {
+    isValid = false;
+  }
+  if (!isValid) {
+    throw new TypeError(`The "regexString" must be a valid regex`);
+  } else {
+    return myRegexObject;
+  }
+};
+
 /**
  * Allows switching between tabs using URL or page title.
  *
  * @example
- * await switchTo('https://taiko.gauge.org/') # switch using URL
- * await switchTo('Taiko') # switch using Title
+ * # switch using URL
+ * await switchTo('https://taiko.gauge.org/')
+ * # switch using Title
+ * await switchTo('Taiko')
+ * # switch using regex URL
+ * await swithTo(regex(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/))
```

Comment:
There's a typo this should be `switchTo`.

The idea is to make this api as transparent to the user. Instead of introducing a `regex` method can't we simply check the parameter type and identify this as a regex?

For example (simplified version) check if a parameter is a regex

```
function switchTo(param) {
 if(param instanceof RegExp) {
 } else {}
}
```

If this check is internal then the user does not have to export the regex method. The can just specify the parameter and this api can be.

```
swithTo(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/)
```

Reply:
Ofcourse. I thought of that too. But now suppose that you want to add this regex functionality into other functions. I think regex() will provide a base for all of the functionalities that accepts regex. 

Consider it as an `before()` kinda thing. You can pre-process your regex before doing the real stuff.

Let me know if this sounds good. Otherwise I don't mind removing it and simply parsing the regex.

Thanks.

Reply:
I agree about the re-suability. But we could do the pre-processing inside the `switchTo` and `closeTab`

The `regex` method here is very different from the other api's as it does not perform an action on the browser. Taiko api's are trying to make the api very specific to actions on the browser.

Also can't what this method does 

```
module.exports.regex = async regexString => {
   var isValid = true;
   var myRegexObject = null;
   try {
     myRegexObject = new RegExp(regexString);
   } catch (e) {
     isValid = false;
   }
   if (!isValid) {
     throw new TypeError(`The "regexString" must be a valid regex`);
   } else {
     return myRegexObject;
   }
 };
```

simply be 

```
regexString instanceof RegExp
```

When the user passes `/test/`? Else, just treat it as a string and use that for matching. This way a tester does not have to remember to use the extra api. When it is an instance of `RegExp` it should be valid to use and match.

Again  the api could be

```
await closeTab(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/)
switchTo(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/)
```

with no imports or extra api for the user to learn. We are always trying to make it easier for the user. Hope this makes sense. Sorry for pushing this!

Reply:
Gotcha! I will push with the changes. :)